### PR TITLE
Deprecate the IAuthenticationHandler property

### PR DESCRIPTION
**Pull request by davidfowl**
#861

### CODE REVIEW

File: src/Microsoft.AspNetCore.Http.Features/Authentication/IHttpAuthenticationFeature.cs [hunk 1]
```diff
@@ -9,6 +10,7 @@ public interface IHttpAuthenticationFeature
     {
         ClaimsPrincipal User { get; set; }
 
+        [Obsolete("This is obsolete and will be removed in a future version. The recommended alternative is to use Microsoft.AspNetCore.Authentication.AuthenticationHttpContextExtensions. See https://go.microsoft.com/fwlink/?linkid=845470.")]
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Http.Features/Authentication/IHttpAuthenticationFeature.cs]: Drop "The recommended alternative is to use Microsoft.AspNetCore.Authentication.AuthenticationHttpContextExtensions."? It's not a direct replacement for this API.

**Comment by davidfowl** [comment]
@HaoK Just the entry points.

**Comment by Tratcher** [comment]
We should at least do AuthenticationManager. That would get any code that was accepting it.

**Comment by HaoK** [comment]
@Tratcher just the class and not all of the methods right?

**Comment by Tratcher** [comment]
At least the class. And at that point why not do the methods too?

**Comment by HaoK** [comment]
How would they ever call these methods without going through context.Authentication?

**Comment by Tratcher** [comment]
`public void DoAuth(AuthenticationManager manager);`

Yes someone would have to call context.Authentication before calling this method, but that may not even be in the same project as a method accepting AuthenticationManager.

**Comment by HaoK** [comment]
So in that usage it would be enough to have it on the class then right?  They'd only have access to the methods via context.Authentication or an explicit type reference

**Comment by Tratcher** [comment]
Right, but at that point why not just do all the methods? It's just some copy-paste?

**Comment by HaoK** [comment]
I'm just trying to avoid the viral spread of obsolete/pragmas, if the methods don't spread pragamas sure...

**Pull request merged**